src/platform/utilities/environment/index.js

```diff
@@ -1,43 +1,80 @@
-    BASE_URL: 'https://preview.va.gov',
-  },
-  vagovprod: {
-    API_URL: 'https://api.va.gov',
+/*
+ * Configuration for the current environment.
+ * @module platform/utilities/environment
+ */
+
+import ENVIRONMENTS from '../../../site/constants/environments';
+
+// eslint-disable-next-line no-undef
+const BUILDTYPE = __BUILDTYPE__;
+
+const ENVIRONMENT_CONFIGURATIONS = {
+  [ENVIRONMENTS.VAGOVPROD]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVPROD,
     BASE_URL: 'https://www.va.gov',
+    API_URL: 'https://api.va.gov',
   },
-  vagovstaging: {
-    API_URL: 'https://staging-api.va.gov',
+
+  [ENVIRONMENTS.VAGOVSTAGING]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVSTAGING,
     BASE_URL: 'https://staging.va.gov',
+    API_URL: 'https://staging-api.va.gov',
   },
-  vagovdev: {
-    API_URL: 'https://dev-api.va.gov',
+
+  [ENVIRONMENTS.VAGOVDEV]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVDEV,
     BASE_URL: 'https://dev.va.gov',
+    API_URL: 'https://dev-api.va.gov',
   },
-  localhost: {
+
+  [ENVIRONMENTS.LOCALHOST]: {
+    BUILDTYPE: ENVIRONMENTS.LOCALHOST,
+    BASE_URL: `http://${location.hostname}:${location.port}`,
     API_URL: `http://${location.hostname}:3000`,
-    BASE_URL: `http://${location.hostname}:3001`,
-  },
-  e2e: {
-    API_URL: `http://localhost:${process.env.API_PORT || 3000}`,
-    BASE_URL: `http://localhost:${process.env.WEB_PORT || 3333}`,
   },
 };
 
-function getEnvironment() {
-  let platform;
-
-  if (location.port === '3001') {
-    platform = 'localhost';
-  } else if (location.port === `${process.env.WEB_PORT || 3333}`) {
-    platform = 'e2e';
-  } else {
-    platform = __BUILDTYPE__;
-  }
+const environment = ENVIRONMENT_CONFIGURATIONS[BUILDTYPE];
+const isPort80 = location.port === '' || location.port === 80;
 
-  return _Environments[platform];
+if (!isPort80) {
+  // It's possible that we're executing a certain build-type under a hostname
+  // other than the host that it's configured. This is the case for integration tests,
+  // where we're testing the vagovprod (production) build-type by serving the production-compiled files,
+  // but on a server running under localhost. This would also be the case if we're testing the
+  // production environment on our local machines.
+  const LOCALHOST_ENV = ENVIRONMENT_CONFIGURATIONS[ENVIRONMENTS.LOCALHOST];
+  environment.API_URL = LOCALHOST_ENV.API_URL;
+  environment.BASE_URL = LOCALHOST_ENV.BASE_URL;
 }
 
-const environment = getEnvironment();
-module.exports = environment;
+export default {
+  /** The name of the environment under which the site is currently executing. */
+  BUILDTYPE: environment.BUILDTYPE,
+
+  /** The address of the FE website configured for this environment. */
+  BASE_URL: environment.BASE_URL,
+
+  /** The address of the API configured for this environment. */
+  API_URL: environment.API_URL,
+
+  /** Determines whether the current environment is a production environment. */
+  isProduction() {
+    return this.BUILDTYPE === ENVIRONMENTS.VAGOVPROD;
```

Comment:
Instead of using `this`, could we do `environment.BUILDTYPE`? I know it's more repetitive, but it's a bit safer because if someone destructures `environment` and pulls a function out, it'll still work correctly, whereas this version won't.

Reply:
Makes perfect sense! I was using `this` only because of this test, https://github.com/department-of-veterans-affairs/vets-website/pull/9162/files#diff-c2e1d1dcef76f34ec88b742ad38d04ffR317, which changes the `BUILDTYPE` property to test in different environments, but I think I can visit that test and see if it can be spruced up.